### PR TITLE
[to #295] Split integrated tests to groups

### LINKED ISSUE #295: cdc: Speed up integrated tests
## Feature Request

**Is your feature request related to a problem? Please describe:**

The integrated test cost almost one hour. It's quite inefficiency for waiting CI to check the correctness of PR.

**Describe the feature you'd like:**

We should try to speed up it.

**Describe alternatives you've considered:**

* Separate `make integration_test` to several sub-targets, and create jenkins steps for each one to run parallel.
  * TiFlow use a groovy ([here](https://github.com/PingCAP-QE/ci/blob/main/jenkins/pipelines/ci/ticdc/integration_test_common.groovy)) to run each test case as a step (see [here](https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_integration_test/detail/cdc_ghpr_integration_test/16502/pipeline/135/)). It is great but we may need some help to do this.
* Combine some test cases to one to reduce the cost of establishing & destroying cluster (refer to [available test](https://github.com/tikv/migration/blob/main/cdc/tests/integration_tests/availability/run.sh)).
* Inspect some long test cases to see is there any change to cut some time.

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by pingyu**
Signed-off-by: Ping Yu <yuping@pingcap.com>

### What problem does this PR solve?

Issue Number: close #295

Problem Description: **Speed up integrated tests**

### What is changed and how does it work?

1. Split integrated test cases to groups.
2. Add README for how to run tests ([rendered](https://github.com/pingyu/migration/blob/it-split-groups/cdc/tests/README.md)).

Usage:
```
tests/integration_tests/run_group.sh [G00|G01| .. |G12|others]
```

### Code changes

- No code

### Check List for Tests

This PR has been tested by at least one of the following methods:
- Integration test
- Manual test (add detailed scripts or steps below)

```bash
# tests/integration_tests/run_group.sh G00
Run cases: autorandom
=================>> Running test /cdc/tests/integration_tests/autorandom/run.sh using Sink-Type: tikv... <<=================
The 1 times to try to start tidb cluster...
start tidb cluster in /tmp/tikv_cdc_test/autorandom
...
...
[Thu Nov 17 10:46:02 UTC 2022] <<<<<< run test case autorandom success! >>>>>>
#
```

```bash
# make integration_test CASE="cli cli_tls"
tests/integration_tests/run.sh "cli cli_tls"
=================>> Running test /cdc/tests/integration_tests/cli/run.sh using Sink-Type: tikv... <<=================
The 1 times to try to start tidb cluster...
start tidb cluster in /tmp/tikv_cdc_test/cli
...
...
[Thu Nov 17 09:39:19 UTC 2022] <<<<<< run test case cli success! >>>>>>
=================>> Running test /cdc/tests/integration_tests/cli_tls/run.sh using Sink-Type: tikv... <<=================
The 1 times to try to start tls tidb cluster...
start tidb cluster in /tmp/tikv_cdc_test/cli_tls
...
...
[Thu Nov 17 09:41:10 UTC 2022] <<<<<< run test case cli_tls success! >>>>>>
\033[0;36m<<< Run all test success >>>\033[0m
```

### Side effects

- No side effects

### Related changes

- No related changes

### To reviewers

Please follow these principles to check this pull requests:

- **Concentration**. One pull request should only do one thing. No matter how small it is, the change does exactly one thing and gets it right. Don't mix other changes into it.
- **Tests**. A pull request should be test covered, whether the tests are unit tests, integration tests, or end-to-end tests. Tests should be sufficient, correct and don't slow down the CI pipeline largely.
- **Functionality**. The pull request should implement what the author intends to do and fit well in the existing code base, resolve a real problem for TiDB/TiKV users. To get the author's intention and the pull request design, you could follow the discussions in the corresponding GitHub issue or [internal.tidb.io](https://internals.tidb.io) topic.
- **Style**. Code in the pull request should follow common programming style. *(For Go, we have style checkers in CI)*. However, sometimes the existing code is inconsistent with the style guide, you should maintain consistency with the existing code or file a new issue to fix the existing code style first.
- **Documentation**. If a pull request changes how users build, test, interact with, or release code, you must check whether it also updates the related documentation such as READMEs and any generated reference docs. Similarly, if a pull request deletes or deprecates code, you must check whether or not the corresponding documentation should also be deleted.
- **Performance**. If you find the pull request may affect performance, you could ask the author to provide a benchmark result.

*(The above text mainly refers to [TiDB Development Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/review-a-pr.html#checking-pull-requests). It's also highly recommended to read about [Writing code review comments](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/review-a-pr.html#writing-code-review-comments))*

### CODE REVIEW

**Comment by pingyu** [comment]
@zeminzhou @haojinming PTAL, thanks~

File: cdc/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+### Integration Test
+
+#### Run integration tests locally
+
+1. Run `make integration_test_build` to generate `tikv-cdc` binaries for integration test
+
+2. Run `make integration_test` to execute the integration tests. This command will
+
+   1. Check that all required executables exist.
```
Comment (zeminzhou) [inline cdc/tests/README.md:48]: `make integration_test` also download binaries that are required.

Comment (pingyu) [reply cdc/tests/README.md]: Fixed, PTAL~

**Review by zeminzhou — APPROVED** [review]
LGTM~

**Review by haojinming — APPROVED** [review]
LGTM

**Pull request merged**